Add New Features page to patterns

heyjoethomas opened the pull request:
Moving the new features/new badge pattern back into the patterns section and out of legacy. New content has also been added for clarity. 

<!--
  ☝️How to write a good PR title:
  - Prefix it with [ComponentName] (if applicable), for example: [Button]
  - Start with a verb, for example: Add, Delete, Improve, Fix…
  - Give as much context as necessary and as little as possible
  - Prefix it with [WIP] while it’s a work in progress
-->
<img width="1624" alt="Screenshot 2023-05-10 at 2 15 22 PM" src="https://github.com/Shopify/polaris/assets/5749440/945100fe-47c0-4238-a17b-8c56ff133095">


### WHY are these changes introduced?
There has been an increasing need for guidance on how we communicate new features in the admin to merchants. Publishing this content is a stop-gap measure as we collect more comprehensive thoughts on the topic across teams. 

Fixes #0000 <!-- link to issue if one exists -->

<!--
  Context about the problem that’s being addressed.
-->

<!--
  Summary of the changes committed.

  Before / after screenshots are appreciated for UI changes. Make sure to include alt text that describes the screenshot.

  If you include an animated gif showing your change, wrapping it in a details tag is recommended. Gifs usually autoplay, which can cause accessibility issues for people reviewing your PR:

    <details>
      <summary>Summary of your gif(s)</summary>
      <img src="..." alt="Description of what the gif shows">
    </details>
-->

<!-- ℹ️ Delete the following for small / trivial changes -->

### How to 🎩

🖥 [Local development instructions](https://github.com/Shopify/polaris/blob/main/README.md#local-development)
🗒 [General tophatting guidelines](https://github.com/Shopify/polaris/blob/main/documentation/Tophatting.md)
📄 [Changelog guidelines](https://github.com/Shopify/polaris/blob/main/.github/CONTRIBUTING.md#changelog)

<!--
  Give as much information as needed to experiment with the component
  in the playground.
-->

<details>
<summary>Copy-paste this code in <code>playground/Playground.tsx</code>:</summary>

```jsx
import React from 'react';
import {Page} from '../src';

export function Playground() {
  return (
    <Page title="Playground">
      {/* Add the code you want to test in here */}
    </Page>
  );
}
```

</details>

### 🎩 checklist

- [ ] Tested on [mobile](https://github.com/Shopify/polaris/blob/main/documentation/Tophatting.md#cross-browser-testing)
- [ ] Tested on [multiple browsers](https://help.shopify.com/en/manual/shopify-admin/supported-browsers)
- [ ] Tested for [accessibility](https://github.com/Shopify/polaris/blob/main/documentation/Accessibility%20testing.md)
- [ ] Updated the component's `README.md` with documentation changes
- [ ] [Tophatted documentation](https://github.com/Shopify/polaris/blob/main/documentation/Tophatting%20documentation.md) changes in the style guide

heyjoethomas committed:
Add New Features page to patterns

Moving the new features pattern back into the patterns section and out of legacy. New content has also been added for clarity.

heyjoethomas committed:
add changeset

sam-b-rose commented on polaris.shopify.com/content/patterns/new-features/index.md:
```diff
@@ -0,0 +1,23 @@
+---
+title: New Features
```
```suggestion
title: New features
```

sam-b-rose reviewed (approved):
Thanks for bringing back this legacy new feature!?! 💯  Just one small casing-related comment.

sophschneider commented on polaris.shopify.com/content/patterns/new-features/index.md:
```diff
@@ -0,0 +1,23 @@
+- See another two-column layout in use in the [Resource detail layout](/patterns/resource-details-layout) pattern.
+- See a single-column layout in use in the [Resource index layout](/patterns/resource-index-layout) pattern.
+- Learn more about [Layout](https://shopify.dev/apps/design-guidelines/layout) in the app design guidelines.
+- Check out the Polaris [Space](/design/space) guidelines to understand Polaris grid and spacing scale.
```
Are layouts or space is related to new feature? or is this copied over from somewhere else?

heyjoethomas replied:
Yes. My bad. Will remove

martenbjork commented on polaris.shopify.com/content/patterns/new-features/variants/default.md:
```diff
@@ -0,0 +1,46 @@
+## Avoid highlighting individual features within a page
+
+As the Shopify admin UI continues to evolve, it's natural for teams to want to draw attention to new features and improvements. However, we strongly advise against using "new" badges or blue pips to highlight these additions within individual sections like Products, Orders, and Customers.
+
```
I think this should live before the previous section. This makes a good intro.

martenbjork commented on polaris.shopify.com/content/patterns/new-features/variants/default.md:
```diff
@@ -0,0 +1,46 @@
+  **Inconsistency** - Different teams might use different methods to highlight features, leading to an inconsistent experience for merchants.
+  
+  **Visual clutter** - Over time, the admin UI may become cluttered with numerous badges and pips, making it difficult for merchants to focus on their primary tasks.
+
```
Formatting: This should probably just be h4s (####)

heyjoethomas replied:
The spacing on the type isn't great this way which is why I had it formatted the way I did. Thoughts? 
<img width="890" alt="Screenshot 2023-05-11 at 9 57 32 AM" src="https://github.com/Shopify/polaris/assets/5749440/e5d60695-053c-4bce-9479-210dfe44312d">

martenbjork replied:
Crikey, that's awful but I'd still ship the h4s. Avoiding the h4s only hides the issue and prevents it from being fixed.

martenbjork commented on polaris.shopify.com/content/patterns/new-features/variants/default.md:
```diff
@@ -0,0 +1,46 @@
+- It’s worth distracting merchants from their current workflow to inform them about a whole new part of Shopify
+
+Things that generally meet this criteria are large and impactful. Examples could be entire new pages in settings or the introduction of an entirely new technology or capability. 
+
```
There is a risk that teams have tunnel vision with regards to their new feature, and think that their feature ticks all these boxes. I think we can phrase this even more strongly.

heyjoethomas replied:
Good idea. Will shift this around.

martenbjork reviewed (approved):
This is great! Really glad you're jumping on this 🔥 I left some comments, but no blockers.

heyjoethomas committed:
Update polaris.shopify.com/content/patterns/new-features/index.md

Co-authored-by: Sam Rose <sam.rose@shopify.com>

heyjoethomas committed:
Update index.md

Removed related resources that were not relevant

heyjoethomas committed:
Reorganize content

Moved a couple paragraphs around and removed first heading to create better flow of information.

martenbjork commented on polaris.shopify.com/content/patterns/new-features/variants/default.md:
```diff
@@ -0,0 +1,47 @@
+
+ #### Inconsistency
+ 
+ Different teams might use different methods to highlight features, leading to an inconsistent experience for merchants.
```
There is a stray space character before this paragraph and the #### before it.

heyjoethomas replied:
🤦‍♂️

heyjoethomas committed:
Fixed formatting

Fixed formatting issues

heyjoethomas committed:
Update default.md

Heading change
